corehq/apps/hqmedia/models.py

```diff
@@ -685,24 +685,33 @@ def create_mapping(self, multimedia, form_path, save=True):
                 updated_doc = self.get(self._id)
                 updated_doc.create_mapping(multimedia, form_path)
 
-    def get_media_objects(self):
+    def get_media_objects(self, languages=None):
         """
             Gets all the media objects stored in the multimedia map.
+            If passed a profile, will only get those that are used
+            in a language in the profile.
         """
         found_missing_mm = False
+        filter_multimedia = languages and self.media_language_map
+        if filter_multimedia:
+            media_list = []
+            for lang in languages:
+                media_list += self.media_language_map[lang].media_refs
```

Comment:
small: would be better to use `media_list.extend(...)` since that would not create a new list on each iteration.